examples/06_assets/main.rs

```diff
@@ -59,62 +87,105 @@ impl State for Example {
-        asset_manager.load_asset_from_data::<Texture, [f32; 4]>("green", [0.0, 0.0, 0.1, 1.0]);
-        asset_manager.load_asset::<Mesh>("Mesh000", "obj");
-        asset_manager.load_asset::<Mesh>("Mesh001", "obj");
+        asset_manager.load_asset_from_data::<Texture, [f32; 4]>("green",     [0.0, 1.0, 0.2, 1.0]);
+        asset_manager.load_asset_from_data::<Texture, [f32; 4]>("tan",       [0.8, 0.6, 0.5, 1.0]);
+        asset_manager.load_asset::<Texture>("crate", "png");
+        asset_manager.load_asset::<Texture>("grass", "bmp");
+
+        // Load/generate meshes
+        asset_manager.load_asset::<Mesh>("teapot", "obj");
+        asset_manager.load_asset::<Mesh>("lid", "obj");
+        asset_manager.load_asset::<Mesh>("cube", "obj");
+        asset_manager.load_asset::<Mesh>("sphere", "obj");
+
+        // Also add custom asset loader and load mesh
+        asset_manager.register_loader::<Mesh, CustomObj>("custom");
+        asset_manager.load_asset::<Mesh>("cuboid", "custom");
+
+        // Add teapot and lid to scene
+        for mesh in vec!["lid", "teapot"].iter() {
+            let mut transform = LocalTransform::default();
+            transform.rotation = [0.5, 0.5, -0.5, -0.5];
+            transform.translation = [5.0, 0.0, 5.0];
+            let renderable = asset_manager.create_renderable(mesh, "dark_blue", "green").unwrap();
+            world.create_now()
+                .with(renderable)
+                .with(transform)
+                .with(Transform::default())
+                .build();
+        }
 
-        let renderable = asset_manager.create_renderable("Mesh000", "dark_blue", "green").unwrap();
+        // Add custom cube object to scene
+        let renderable = asset_manager.create_renderable("cuboid", "dark_blue", "green").unwrap();
+        let mut transform = LocalTransform::default();
+        transform.translation = [-5.0, 0.0, 0.0];
+        transform.scale = [2.0, 2.0, 2.0];
         world.create_now()
             .with(renderable)
+            .with(transform)
             .with(Transform::default())
             .build();
 
-        let renderable = asset_manager.create_renderable("Mesh001", "dark_blue", "green").unwrap();
+        // Add cube to scene
+        let renderable = asset_manager.create_renderable("cube", "crate", "tan").unwrap();
+        let mut transform = LocalTransform::default();
+        transform.translation = [5.0, 0.0, 0.0];
+        transform.scale = [2.0, 2.0, 2.0];
         world.create_now()
             .with(renderable)
+            .with(transform)
             .with(Transform::default())
             .build();
 
+        // Add sphere to scene
+        let renderable = asset_manager.create_renderable("sphere", "grass", "green").unwrap();
+        let mut transform = LocalTransform::default();
+        transform.translation = [-5.0, 0.0, 7.5];
+        transform.rotation = [0.70711, 0.70711, 0.0, 0.0];
```

Comment:
Is this correct? The rotation is a quaternion [w, x, y, z] so I'm thinking this might need to be `[1.0, 0.70711, 0.70711, 0.0]` if you wanted a 45 degree angle on x and y.